tools/flaky-test-reporter/github_issue.go

```diff
@@ -174,30 +179,35 @@ func (gih *GithubIssueHandler) createHistoryUnicode(rd RepoData, comment, testFu
 		currentUnicode += fmt.Sprintf(" [%s](%s)", statusUnicode, url)
 	}
 
-	oldHistory := regexp.MustCompile(reHistory).FindStringSubmatch(comment)
-	res = fmt.Sprintf("\n%s", currentUnicode)
-	if len(oldHistory) >= 2 {
-		oldHistoryEntries := strings.Split(oldHistory[1], "\n")
-		if len(oldHistoryEntries) >= maxHistoryEntries {
-			oldHistoryEntries = oldHistoryEntries[:(maxHistoryEntries - 1)]
-		}
-		res += strings.Join(oldHistoryEntries, "\n")
+	// Make sure there is no dupe of records
+	uniqHistoryEntries := sets.String{}
+	oldHistory := regexp.MustCompile(reSingleRecord).FindAllStringSubmatch(comment, -1)
+	for _, hist := range oldHistory {
+		// Remove <!------End of History------>" that were introduced on some lines.
+		// TODO(chaodaiG): removing this logic once this is all cleared in
+		// existing bugs.
+		uniqHistoryEntries.Insert(strings.ReplaceAll(hist[0], afterHistoryToken, ""))
+	}
+	records := append([]string{currentUnicode}, uniqHistoryEntries.List()...)
+	sort.Slice(records, func(i, j int) bool {
+		return records[i] > records[j]
+	})
+	// Truncate to keep only 100 records, as there is a 65535 characters limit,
+	// see https://github.com/knative/test-infra/issues/1326
+	if len(records) > 100 {
+		records = records[:100]
```

Comment:
Are you positively sure each record is shorter than 655 characters, plus any header/footer for a 100-item entry? If so, better make it clear here.

Reply:
good point, updated